general/mem_manager.hpp

```diff
@@ -600,6 +604,13 @@ class MemoryManager
                          MemoryType h_mt, MemoryType d_mt,
                          bool own, bool alias, unsigned &flags);
 
+   /** @brief If true, class Memory will register the base Memory when creating
+       aliases with Memory<T>::MakeAlias(). */
+   /** This value is initialized during configuration: it is set to true when
+       the default device memory type is not the same as the default host memory
+       type. */
+   static bool RegisterAliasBases() { return register_alias_bases; }
```

Comment:
```suggestion
   static bool RegisterAliasBases() { return host_mem_type != device_mem_type; }
```
and remove `static bool register_alias_bases;`.

Reply:
I thought, in the future, we may want to tweak the logic for when to set this flag -- that's why I used a separate variable. However, right now, your suggestion is equivalent, so I'll just use that -- I don't have a strong preference either way.